Refactoring and Applying Design-Patterns

Tigerfriend1 opened the pull request:
Hello!

I am impressed by your open source PhotonCamera project.

So we just want to contribute a little to the project.

Six students from Pusan National University in Korea, including me, refactored a part of the project.

Without modifying the functionality, we focused on refactoring and applying design patterns. We have confirmed that the project functions normally through our own tests.

Detailed descriptions of each refactoring and design pattern application can be found in each commit.

The test devices we used were Xiaomi Redmi Note 7, Galaxy 10, and Galaxy 21.

Additionally, we have provided test cases for some of the refactored classes.

Thank you for reading!

---

Based on the [SOLID-principle](https://en.wikipedia.org/wiki/SOLID). 
Major Refactoring: Variable renaming, Eliminate code duplication
Major Design Patterns: State pattern, Strategy pattern, Factory pattern

NAINSOO committed:
refactor: DebugClient.java

I refactored the constructor of the debugClient using the extract method, so the receiveMessageFromServer method, the sendMessage method, and the receiveMessage method were added.

NAINSOO committed:
refactor: DebugClient.java

i refactored the getObjectString method using the extract method. so the getClassName method is added.

Hooni07 committed:
refactor: add findMaxValue() method and apply drawArray

Hooni07 committed:
refactor: add findMaxValue() method and apply drawArray

Hooni07 committed:
refactor: add drawGraph() method and apply drawArray

Hooni07 committed:
Refactor: apply findMaxValue() and drawGraph()

Hooni07 committed:
modify: adjust unused variables in drawArray

Hooni07 committed:
refactor: modify linearRegressionC()

Yoongunwo committed:
horizontalpicker method refact
HorizontalPicker(Context context, AttributeSet attributeSet, int defStyle) 생성자 refactor
Refactor : HorizontalPicker 생성자 refactoring

attris -> attributeSet : naming refactor
- 변수 명이 파악하기 힘듬
a -> typedArray : naming refactor
- 변수 명이 파악하기 힘듬
selectEllipsize() : extract method
- SIP 불충족
setBorginMetrics() :  extract method
- SIP 불충족
initializeConstants() : extract method
- SIP 불충족

Yoongunwo committed:
주석 제거

Hooni07 committed:
modify: increased readability

NAINSOO committed:
refactor: DebugClient.java

입력 텍스트 전체 삭제102 / 5000듣기복사
번역하기
The switch statement in the setKey method of DebugClient.java was replaced with polymorphism, so an interface requestKeyType used for polymorphism was added.

MinWoo-Seong committed:
Refactor: GLBasePipeline

extract duplicate code from runAll() and runAllRaw()

renaming startT, endT

Seong-MinWoo committed:
Revert "Refactor: GLBasePipeline"

This reverts commit b6a673549d4942272227c050e55921948c037c3e.

Seong-MinWoo committed:
Refactor: GLBasePipeline

Hooni07 committed:
refactor: add setWallPaint(), drawCanvas() method and apply drawArray()

Hooni07 committed:
modify: change method's creation order(setWallPaint, drawCanvas)

sm136599 committed:
refactor: improve clarity and readability of chooseOptimalSize()

sm136599 committed:
refactor: improve clarity and efficiency of getCameraOutputSize(Size[])

- remove unnecessary list conversion

sm136599 committed:
Revert "refactor: improve clarity and efficiency of getCameraOutputSize(Size[])"

This reverts commit 3c39e494b731b91c76e9134213ea4f229da03add.

sm136599 committed:
refactor: improve clarity and efficiency of getCameraOutputSize(Size[])

- remove unnecessary list conversion

sm136599 committed:
refactor: improve clarity and efficiency of getCameraOutputSize(Size[], Size)

- remove unnecessary list conversion

Tigerfriend1 committed:
Merge pull request #1 from Yoongunwo/refactor/yoon

Refactor/yoon
photoncamera/ui/camera/views/modeswitcher/wefika/horizontalpicker/HorizontalPicker Class refactoring

Hooni07 committed:
Refactor: make isBetween() method to improve the expression in 'if' conditions

Hooni07 committed:
modify: change method name(isBetween -> orientationPosition)

MinWoo-Seong committed:
Refactor: ColorCorrectionTransform

Tigerfriend1 committed:
Merge pull request #2 from Hooni07/refactor

(Hooni07)Refactor: refactoring Utilities.java and CustomOrientationEventListener.java

Tigerfriend1 committed:
Merge pull request #4 from MinWoo-Seong/dev

 Refactor: GLBasePipeline

Yoongunwo committed:
tigerfriend head

tigerfriend head

Yoongunwo committed:
Refactor : ParamController.java Rename

update() 메서드
Observable o -> observable : rename
Object obj -> object : rename

MinWoo-Seong committed:
Refactor: ColorCorrectionTransform

Hooni07 committed:
refactor: add 'convertToFloatArray()' and modify code

NAINSOO committed:
Refactor: DebugClient.java

using factory method, I moved a switch statement into the Command interface's static method. The Command interface is maked by me for strategy pattern in the ParseControl method. And, i refactor  requestKeyType that i maked before
Too.

Yoongunwo committed:
Refactor : SettingsBarEntryProviderFactory, Factory Method Pattern

Factory Method Pattern 적용
Open-close principle을 충족시키기 위해서 적용
다양한 SettingsBarEntry를 생성하기 위해서 적용하게 되었다.

현재는 SettingsBarEntryProvider만 있지만
이를 추상 클래스로 만들고 여러가지 버전의 SettingsBar를 생성하고자
하였다.

Yoongunwo committed:
Refactor SettingsBarEntryProviderFactory

Yoongunwo committed:
Refactor : SettingsBarEntryProvider - Factory Method Pattern

Factory Method Pattern을
적용시키기 위해 추상 클래스인
SettingsBarEntryProviderVer1 구현

Yoongunwo committed:
Merge branch 'refactor/yoon' of https://github.com/Yoongunwo/PhotonCamera into refactor/yoon

Yoongunwo committed:
Refactor/yoon

Refactor/yoon

sm136599 committed:
Minor: typo fix

fillInCameraCharateristics -> fillInCameraCharacteristics

sm136599 committed:
Minor: Organize code by moving initialization to static blocks

sm136599 committed:
Refactor: Extract method from Switch

 - mCaptureCallback's method process
 - extract method previewProcess(), waitingLockProcess(), waitingPrecaptureProcess(), waitingNonPrecaptureProcess()

Tigerfriend1 committed:
Merge pull request #6 from Hooni07/refactor

(Hooni07)Refactor: refactoring Utilities.java

Hooni07 committed:
Refactor: apply strategy method pattern(add interface, class and modify Utilities.java)

NAINSOO committed:
refactor: DebugClient.java

I renamed the interface name of requestKeyType into RequestKeyType for camel case. and it's subclass name too.
 The inheritance classes inside the Command interface and the RequestKeyType interface were separated into their own files.

NAINSOO committed:
refactor: Command.java and RequestKeyType.java

i replace string type in the switch statement with enum class.
so TypeEnum.java and KewWord.java fille which have an enum class were added.

Tigerfriend1 committed:
refactor: Apply the State Pattern to modify the value settings based on the mode.

Create a currentState object and executes the reConfigureModeViews function based on the camera mode

Tigerfriend1 committed:
refactor: create the "CameraModeState" interface to utilize the State Pattern.

Within CameraModeState, the appropriate reConfigureModeViews function is executed based on the camera mode.

Seong-MinWoo committed:
Merge branch 'refactor' into dev

Tigerfriend1 committed:
Merge pull request #5 from sm136599/refactor

refactor: CaptrueController

Tigerfriend1 committed:
Merge pull request #7 from NAINSOO/refactor

refactor: DebugClient.java

sm136599 committed:
Refactor: Reduced Complexity in createCameraPreviewSession()

- Extract method configureSurfaces() and setCaptureRequestBuilder()

Tigerfriend1 committed:
Merge pull request #8 from Hooni07/refactor

(Hooni07)Refactor: apply Strategy Method

NAINSOO committed:
refactor : Debugger.java

Since the constructor of the file violates SRP, it was refactored using the Extract Method.

Seong-MinWoo committed:
Refactor: DefaultImageSaver

extract image save method

Tigerfriend1 committed:
Merge pull request #10 from NAINSOO/refactor

refactor : Debugger.java

Seong-MinWoo committed:
Refactor: DefaultImageSaver

Tigerfriend1 committed:
refactor: CameraFragment, CameraUIView

In order to satisfy the SOLID principle, the inner class and interface were separated to the outside.

1. Detach the CameraUIEventsListener interface from within the CameraUIView interface to the outside(ISP)
2. Extract the CameraUIController from the CameraFragment class to the outside(SRP)

Now, When modifying features, it became clear which classes should be accessed.

Yoongunwo committed:
Fix : HorizontalPicker.java

final field error fix

Yoongunwo committed:
Fix : HorizontalPicker.java

Fix : HorizontalPicker.java

Yoongunwo committed:
Refactor : HorizontalPicker 생성자 refactoring

getTestPaint() : extract method
- SIP 불충족

Yoongunwo committed:
Fix HorizontalPicker.java

에러 수정

Yoongunwo committed:
Refactor : HorizontalPicker 

Refactor/yoon

Tigerfriend1 committed:
Merge pull request #12 from Yoongunwo/dev

Design Pattern 및 Refactoring 적용

Tigerfriend1 committed:
Merge pull request #13 from Tigerfriend1/mergeTest

Merge test finish(pull request #12 from Yoongunwo/dev)

Tigerfriend1 committed:
Merge pull request #9 from sm136599/refactor

Refactor: CaptureController.java

Tigerfriend1 committed:
Merge pull request #14 from Tigerfriend1/mergeTest

Merge test finish(Refactor: CaptureController.java #9)

Tigerfriend1 committed:
Merge pull request #11 from MinWoo-Seong/dev

Refactor: DefaultImageSaver

Tigerfriend1 committed:
Merge pull request #15 from Tigerfriend1/mergeTest

Merge test finish(Refactor: DefaultImageSaver #11)

Yoongunwo committed:
Fix : SettingsBarEntryProvider

SettingsBarEntryProvider 클래스를 사용하는 부분을
SettingsBarEntryProviderFactory 클래스 적용

Yoongunwo committed:
Fix : SettingsBarEntryProvider

Fix : SettingsBarEntryProvider

NAINSOO committed:
issue : Debugger.java

I solved the app execution problem caused by NullPointerException by adding a null processing if statement to Debugger.java.

Tigerfriend1 committed:
Merge pull request #16 from NAINSOO/refactor

issue : Debugger.java

Tigerfriend1 committed:
Merge branch 'mergeTest' into dev

Tigerfriend1 committed:
Merge pull request #18 from Yoongunwo/dev

Fix : CameraFragment.java

Tigerfriend1 committed:
fix: SettingsBarEntryProvider

Change SettingsBarEntryProvider to SettingsBarEntryProviderVer1fix: SettingsBarEntryProvider

Change SettingsBarEntryProvider to SettingsBarEntryProviderVer1fix: SettingsBarEntryProvider

Change SettingsBarEntryProvider to SettingsBarEntryProviderVer1

Tigerfriend1 committed:
Merge branch 'refactor' into mergeTest

Tigerfriend1 committed:
Merge pull request #19 from Tigerfriend1/mergeTest

Merge test finish(Fix : SettingsBarEntryProvider#17 )

Yoongunwo committed:
Testing : RANSACTest.java

RANSAC class의 perform() 메서드를
statement Coverage하는 클래스이다.

perform() 메서드에서 RANDOM 클래스의 random() 메서드를
호출하기 때문에 예상 결과값은 알 수 없지만
전체 코드가 실행 가능한지 testing하였다.
실행 결과 100%의 statement coverage를 보여줬다.

Yoongunwo committed:
Testing : DepthPageTransformerTest.java

DepthPageTransformer class의 transformPage() 메서드를
2-value boundary value analysis를 수행하였다.

VEC-1으로서 condition은 [-Infinity, -1) 로서
입력 값으로서 -2를 넣어 주었다.
-1을 안 넣은 이유는 VEC-2에 추가하기 때문에 생략하였다.

Yoongunwo committed:
Refactor/yoon

Refactor/yoon

Hooni07 committed:
test: add 'UtilitiesTest.java' and 'CustomOrientationEventListenerTest.java'

Hooni07 committed:
refactor: add test case in 'UtilitiesTest.java' and little modification in 'Utilities.java'

Hooni07 committed:
test: add test case in 'UtilitiesTest.java', 'CustomOrientationEventListener.java'

Hooni07 committed:
test: add test case in 'CustomOrientationEventListener.java'

Hooni07 committed:
edit: modify code

Hooni07 committed:
test: add test case in 'Utilities.java'

Hooni07 committed:
edit: change setting on 'build.gradle'

Tigerfriend1 committed:
revert build.gradle

Tigerfriend1 committed:
Merge pull request #20 from Hooni07/refactor

Test: Junit Testing 'Utilities.java' and 'CustomOrientationEventListener.java'

Yoongunwo committed:
Testing : DepthPageTransformerTest.java

DepthPageTransformer class의 transformPage() 메서드를
2-value boundary value analysis를 수행하였다.

VEC-2으로서 condition은 [-1, 0]이다.
따라서 입력 값으로 -1, 0을 넣어주었다.
-2, 1을 안 넣은 이유는 VEC-1, VEC-3에 추가하기 때문에 생략하였다.

Yoongunwo committed:
Testing : DepthPageTransformerTest.java
DepthPageTransformer class의 transformPage() 메서드를
2-value boundary value analysis를 수행하였다.

VEC-3으로서 condition은 (0, 1] 로서
입력 값으로서 0.1, 1을 넣어 주었다.
0보다 큰 실수로서 경계값을 0.1을 잡아주었다.
0, 2를 안 넣은 이유는 VEC-2, VEC-4에 추가하기 때문에 생략하였다.

Yoongunwo committed:
Testing : DepthPageTransformerTest.java
DepthPageTransformer class의 transformPage() 메서드를
2-value boundary value analysis를 수행하였다.

VEC-4으로서 condition은 (1, +Infinity]이다.
입력 값으로서 2를 넣어 주었다.
1을 안 넣은 이유는 VEC-3에 추가하기 때문에 생략하였다.

Yoongunwo committed:
Refactor/yoon

DepthPageTransformer class의 transformPage() 메서드를
2-value boundary value analysis를 수행하였다.

VEC-4으로서 condition은 (1, +Infinity]이다.
입력 값으로서 2를 넣어 주었다.
1을 안 넣은 이유는 VEC-3에 추가하기 때문에 생략하였다.

NAINSOO committed:
test : DebugClientTest.java

I added a method called testSetkey in the DebugClientTest class to test setKey, a method of DebugClient.
The test confirmed that IllegalArgumentException was caught and modified the RequestKeyType interface to integrate into IllegalStateException.

NAINSOO committed:
test : DebugClientTest.java

I added a method called testGetObjectString in the DebugClientTest class to test DebugClient's method getObjectString.

MinWoo-Seong committed:
Test: Junit Testing ColorCorrectionTransform

Purpose: Read String to Matrix, Matrixes, Cube, Cubes
Input : String
Excpected : assertEqual

Purpose: combine value of matrix
Input : float[]
Expected : assertCombine = output of combineMatrix

sm136599 committed:
Test: test method chooseOptimalSize()

Purpose: chooseOptimalSize from several sizes and test for line coverage
TestCase1
Input: sizes {(1920, 1080)}
Expected: (1920, 1080)
 TestCase2
 Input: sizes {(1980, 1080), (1280, 720), (1600, 900)}
Expected: (1600, 900)
TestCase3
Input: sizes {(1600, 900), (1280, 720)}
Expected: (1600, 900)
TestCase4
Input: sizes {(2560, 1440)}
Expected: (2560, 1440)

sm136599 committed:
Minor: Add method for test

Add method getCameraOutputSizeTest(Size[]) and getCameraOutputSizeTest(Size[], Size) for test method getCameraOutputSize(Size[]), getCameraOutputSize(Size[], Size)

sm136599 committed:
Test: test method getCameraOutputSize(Size[])

Purpose: test method getCameraOutputSize(Size[]) with line coverage
TestCase1 - for normal 
Input: sizes {(1980, 1080), (1280, 720), (1600, 900)}
Expected: (1920, 1080)
TestCase2  - for largest Size > highResolution 
Input: sizes {(1980, 1080), (1280, 720), (8000, 4500)}
Expected: (1920, 1080)
TestCase3 - for input size == 1 and largest size > highResolution 
Input: sizes {(8000, 4500)}
Expected: null

sm136599 committed:
Minor: Add method for test

sm136599 committed:
Test: test method getCameraOutputSize(Size[], Size)

Purpose: test method getCameraOutputSize(Size[], Size) with line coverage
@TestCase1 - for normal  
Input: sizes {(1980, 1080), (1280, 720), (1600, 900)}
Expected: (1920, 1080)
Note: This test may fail as it appears to be a static test, but the method requires the app to be turned on.

Tigerfriend1 committed:
Merge pull request #25 from Tigerfriend1/mergeTest

Merge test finish(Test: Junit Testing 'Utilities.java' and 'CustomOrientationEventListener.java' #20)

Tigerfriend1 committed:
Merge pull request #21 from NAINSOO/refactor

test : DebugClienTest.java

Tigerfriend1 committed:
Merge pull request #26 from Tigerfriend1/mergeTest

Merge test Finish(test : DebugClienTest.java #21)

Tigerfriend1 committed:
Merge pull request #22 from MinWoo-Seong/dev

Test: Junit Testing ColorCorrectionTransform

Tigerfriend1 committed:
Merge pull request #27 from Tigerfriend1/mergeTest

Merge test finish(Test: Junit Testing ColorCorrectionTransform #22)

Tigerfriend1 committed:
Merge branch 'mergeTest' into dev

Tigerfriend1 committed:
Merge pull request #24 from sm136599/refactor

Test: CaptureController.java

Tigerfriend1 committed:
Merge pull request #28 from Tigerfriend1/mergeTest2

Merge test finish(Test: CaptureController.java #24)

Tigerfriend1 committed:
test: TestSwitchMode.java and TestSwitchModeTest.java

Purpose: Test StatePattern applied successfully.
result : 100% TestSwitchMode coverage achieved.

Tigerfriend1 committed:
Merge pull request #29 from Yoongunwo/dev

Testing : Yoon

Tigerfriend1 committed:
Merge pull request #30 from Tigerfriend1/mergeTest

Merge test finish(Testing : Yoon #29)

Tigerfriend1 committed:
init: default setting

eszdman commented:
Thanks for your work I appreciate it, but I need time to check the correctness of the work, I already found some bugs related to the interface, for example, the rotation of the interface does not work properly (too sensitive for one slope and not sensitive for another) after your improvements.

For testing, you can send preliminary versions to the testing group in a telegram: https://t.me/PhotonCameraEN

Tigerfriend1 commented:
First, I appreciate your feedback. 

Feel free to take enough time for thorough verification, and we don't intend to impose our result on you. If desired, you are also welcome to make modifications or deletions as necessary.

I will send the entire file through Telegram. 
And you can also see the entire file in https://github.com/Tigerfriend1/PhotonCamera

Tigerfriend1 committed:
fix : Fixed interface shaking and showing bug

1. Fixed a bug where the interface was shaking unstable.
2. Fixed an hdrx menu issue that should not be shown
